llama_index/embeddings/huggingface_intel.py

```diff
@@ -0,0 +1,153 @@
+                "`pip install optimum[exporters] optimum-intel neural-compressor`."
+            )
+
+        self._model = model or INCModel.from_pretrained(folder_name)
+        self._tokenizer = tokenizer or AutoTokenizer.from_pretrained(folder_name)
+        self._device = device or infer_torch_device()
+
+        if max_length is None:
+            try:
+                max_length = int(self._model.config.max_position_embeddings)
+            except Exception:
+                raise ValueError(
+                    "Unable to find max_length from model config. "
+                    "Please provide max_length."
+                )
+
+        if pooling not in ["cls", "mean"]:
+            raise ValueError(f"Pooling {pooling} not supported.")
+
+        super().__init__(
+            embed_batch_size=embed_batch_size,
+            callback_manager=callback_manager,
+            folder_name=folder_name,
+            max_length=max_length,
+            pooling=pooling,
+            normalize=normalize,
+            query_instruction=query_instruction,
+            text_instruction=text_instruction,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "INCEmbedding"
+
+    def _mean_pooling(self, model_output: Any, attention_mask: Any) -> Any:
+        """Mean Pooling - Take attention mask into account for correct averaging."""
+        import torch
+
+        # First element of model_output contains all token embeddings
+        token_embeddings = model_output[0]
+        input_mask_expanded = (
+            attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
+        )
+        return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(
+            input_mask_expanded.sum(1), min=1e-9
+        )
+
+    def _cls_pooling(self, model_output: list) -> Any:
+        """Use the CLS token as the pooling token."""
+        if isinstance(model_output, dict):
+            token_embeddings = model_output["last_hidden_state"]
+        else:
+            token_embeddings = model_output[0]
+        return token_embeddings[:, 0]
+
+    def _embed(self, sentences: List[str]) -> List[List[float]]:
+        """Embed sentences."""
+        encoded_input = self._tokenizer(
+            sentences,
+            padding=True,
+            max_length=self.max_length,
+            truncation=True,
+            return_tensors="pt",
+        )
+        import torch
+
+        with torch.inference_mode(), torch.cpu.amp.autocast():
+            model_output = self._model(**encoded_input)
+
+        if self.pooling == "cls":
+            embeddings = self._cls_pooling(model_output)
+        else:
+            embeddings = self._mean_pooling(
+                model_output, encoded_input["attention_mask"].to(self._device)
+            )
+
+        if self.normalize:
+            embeddings = torch.nn.functional.normalize(embeddings, p=2, dim=1)
```

Comment:
should we make `p` `dim` as part of class variable

Reply:
I just took the same [code](https://github.com/danielfleischer/llama_index/blob/intel/llama_index/embeddings/huggingface.py#L172-L175) as in a general huggingface local model.